### PR TITLE
[TFLite][toco] Fix L2_NORMALIZATION incorrectly emitted for axis=None

### PR DESCRIPTION
**Pull request by Niteshagarwal01**
## Problem

`tf.math.l2_normalize(x, axis=None)` produces incorrect results after
TFLite conversion. TFLite normalizes each innermost sub-vector independently
instead of normalizing the whole tensor as one flat vector.

## Root Cause

`identify_l2_normalization.cc` pattern-matches `Square→Sum→Rsqrt→Mul`
and replaces it with `L2NormalizationOperator` WITHOUT checking the
reduction axis of the Sum op.

TFLite's `L2_NORMALIZATION` kernel hardcodes last-axis normalization
(`trailing_dim = DimensionsCount() - 1`). When `axis=None`, `tf.reduce_sum`
reduces over ALL axes. The pattern still matched, emitting the wrong op.

## Fix

Add an axis guard: only emit `L2NormalizationOperator` when the ReduceSum
reduces along the last axis only. For `axis=None` or any other axis,
return early and leave the subgraph as primitive arithmetic ops.

## Tested

- Existing last-axis normalization behavior is unchanged
- `axis=None` case now correctly falls through to arithmetic ops

### CODE REVIEW

**Comment by keerthanakadiri** [comment]
Hi @Niteshagarwal01, Can you please sign the CLA? Thank you!

**Comment by Niteshagarwal01** [comment]
I have signed in my cla can you look again

**Comment by Niteshagarwal01** [comment]
any update regarding my pull request

**Comment by keerthanakadiri** [comment]
Hi @Niteshagarwal01, A review request has been submitted for this PR. Once approved, it will be ready for the next stage of the process. Thank you!

**Comment by Niteshagarwal01** [comment]
Any update regarding merge ?

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
Hi @Niteshagarwal01,
  
  Thank you for contributing this fix to prevent L2_NORMALIZATION from being incorrectly emitted when the reduction axis is `None` or not the last axis. The core logic is correct, but there are a few issues that need to be addressed before this is ready to be merged:
  
  1. **Safety Check on Shape Propagation:**
     Inside `identify_l2_normalization.cc`, calling `model->GetArray(div_or_mul_op->inputs[0]).shape().dimensions_count()` without verifying if shape information has already been propagated can cause a crash (assertion failure inside the shape helper). Please wrap it in a `has_shape()` guard:
     ```cpp
     const auto& input_array = model->GetArray(div_or_mul_op->inputs[0]);
     if (!input_array.has_shape()) {
       return absl::OkStatus();
     }
     const int rank = input_array.shape().dimensions_count();
  
  ```
**2. Revert version.h change:** Bumping TF_MINOR_VERSION in version.h to 2.21 is not appropriate for a bug-fix PR, as version bumps are managed automatically or during standard release steps. Please revert the changes made to tensorflow/lite/version.h.

**3. Add Unit Tests:** We need to add unit tests in third_party/tensorflow/lite/toco/graph_transformations/tests/identify_l2_normalization_test.cc to verify:

**Positive case:** A single trailing axis reduction (e.g. axis={1} on shape {3, 4}) successfully optimizes to a fused L2NormalizationOperator.
**Negative case:** A single non-trailing axis reduction (e.g. axis={0} on shape {3, 4}) leaves the graph unchanged.
**Negative case:** A multi-axis reduction (e.g. axis={0, 1} on shape {3, 4}) leaves the graph unchanged.
